Workshoppers deployment to NPM

kalinchernev opened the pull request:
Relates to issue #2 - feel free to give comments and suggestions.

kalinchernev committed:
Suggest a deployment document

martinheidegger commented on DEPLOYMENT.md:
```diff
@@ -0,0 +1,49 @@
+# [DRAFT] Deployment of workshoppers
```
_Note: PR's are all drafts I believe, :wink: No need to mention it._

martinheidegger commented on DEPLOYMENT.md:
```diff
@@ -0,0 +1,49 @@
+
+Please be kind to prepare the following prerequisites before asking or publishing your package to npm:
+
+- New version of the package should follow semantic versioning
```
How should I provide the version? certainly not in the git repo, right? If the version should be semantic, how to note the changes?

kalinchernev replied:
I was imagining making a release in the github repository. ([example](https://gist.github.com/kalinchernev/1baab65db80984430893754af17def8e))
The release notes can be included manually while creating a draft or automatically if the commits included follow a convention.

martinheidegger commented on DEPLOYMENT.md:
```diff
@@ -0,0 +1,49 @@
+
+Assuming you want to have a deployment of your package to npm:
+- Please read and prepare the prerequisites mentioned in the "Requirements" section
+- Create a request ticket on the board for [deployment requests](https://github.com/workshopper/org/projects)
```
https://github.com/workshopper/org/projects/1

martinheidegger commented on DEPLOYMENT.md:
```diff
@@ -0,0 +1,49 @@
+
+### Travis CI
+- Free service for open source projects
+- The repository should shall a green badge when you embed the following markup in your README.md in the root folder of the workshopper github repository
```
"should shall", seems like a typo

martinheidegger commented on DEPLOYMENT.md:
```diff
@@ -0,0 +1,49 @@
+- Should be named as CHANGELOG or CHANGELOG.md
+- Should reside in the root of the master of the github repository
+- Should contain brief information about the changes done between releases
+- Could use an automated tool such as [standard-version](https://github.com/conventional-changelog/standard-version) or
```
The sentence finishes with "or"

martinheidegger reviewed (changes requested):
@kalinchernev Thank you for preparing the document, I think it could use a little bit clearer language but else it seems okay. @workshopper/deploy What do you think?

kalinchernev committed:
Language cleaning

martinheidegger commented on DEPLOYMENT.md:
```diff
@@ -1,10 +1,12 @@
+# Deployment of workshoppers
 
-Suggesting steps to organize the process of publishing changes from workshoppers to npm.
+Below you can find details regarding the deployment process of [workshoppers on npm](https://www.npmjs.com/org/workshopper). 
```
Suggestion for rephrasing _(only take it if you feel its better)_: 
"This document should help you when deploying new workshopper tutorials [to npm](https://www.npmjs.com/org/workshopper) and on github".

kalinchernev replied:
Sounds better - thanks. Please, feel free to apply further modifications to other phases too, I will not feel bad about it, and it will be easier and faster for you.

martinheidegger commented on DEPLOYMENT.md:
```diff
@@ -1,10 +1,12 @@
+In the table below you will find an overview of package versioning
+between latest release in github and npm.
 
 workshopper | NPM version   | Master of workshopper | Build | NodeJs LTS compatible |
```
How do we establish _"Master of workshopper"_ (it is missing in the STRUCTURE.md)? and _"NodeJs LTS compatible"_? Also "Node.js" [see TM Policy](https://nodejs.org/static/documents/trademark-policy.pdf).

kalinchernev replied:
"Master of workshopper" === "Master branch of github repository of workshopper"
I welcome ideas. I'm not sure how it relates to "NodeJs LTS compatible" to be honest.

martinheidegger commented on DEPLOYMENT.md:
```diff
@@ -13,37 +15,42 @@ javascripting | 2.6.1 | [2.6.1](https://github.com/workshopper/javascripting) |
-- Ensure that the github repository of the workshopper is correctly setup and working with [Travis CI](http://travis-ci.org/).
-- Master branch of workshopper github repository is "buildable", i.e. not broken (Tests are passing, and the lint and other code standards changes are acceptable)
-- Comprehensible change log is provided on the workshopper github master branch in a CHANGELOG.md file. This could be made manually while making a new release on the repository, or prepared automatically from a script (read further below for suggestions)
+- New version of the package should follow [semantic versioning](http://semver.org/);
```
Not a grammar master here but shouldn't the list items be terminated with a `.`?

kalinchernev replied:
Agree, this must be a bad habit of terminating a line of code. Changed to full stops.

martinheidegger commented on DEPLOYMENT.md:
```diff
@@ -13,37 +15,42 @@ javascripting | 2.6.1 | [2.6.1](https://github.com/workshopper/javascripting) |
-- Master branch of workshopper github repository is "buildable", i.e. not broken (Tests are passing, and the lint and other code standards changes are acceptable)
-- Comprehensible change log is provided on the workshopper github master branch in a CHANGELOG.md file. This could be made manually while making a new release on the repository, or prepared automatically from a script (read further below for suggestions)
+- New version of the package should follow [semantic versioning](http://semver.org/);
+- Ensure project github repository of the workshopper is correctly setup and working with [Travis CI](http://travis-ci.org/);
```
"that the" seemed fine to me?!

julianduque reviewed (approved):
LGTM

martinheidegger reviewed (changes requested):
Some small changes. (maybe better to read over it once more?)

kalinchernev committed:
Update table workshoppers

kalinchernev commented:
Hi all,
I updated the document taking latest feedback into consideration. Please be welcome to make further tweaks where you feel necessary.
The table of workshoppers and their differences between github and npm is also updated. Removed the LTS as it's causing confusion. (and I'm not sure it's necessary at this stage)

martinheidegger reviewed (approved):
Thanks for the changes! Merged ^_^

kalinchernev commented:
@martinheidegger thanks for the thumbs up!
I just checked the end result and I see that I have to fix links - will do that today!
